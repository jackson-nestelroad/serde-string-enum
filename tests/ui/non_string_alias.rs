use serde_string_enum::{
    DeserializeLabeledStringEnum,
    SerializeLabeledStringEnum,
};

#[derive(SerializeLabeledStringEnum, DeserializeLabeledStringEnum)]
enum Type {
    #[string = "Grass"]
    Grass,
    #[string = "Fire"]
    #[alias = 1234]
    Fire,
    #[string = "Water"]
    Water,
}

fn main() {}
