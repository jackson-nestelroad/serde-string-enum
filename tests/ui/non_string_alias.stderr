error: "alias" attribute must be a string literal
 --> tests/ui/non_string_alias.rs:6:10
  |
6 | #[derive(SerializeLabeledStringEnum, DeserializeLabeledStringEnum)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `SerializeLabeledStringEnum` (in Nightly builds, run with -Z macro-backtrace for more info)

error: "alias" attribute must be a string literal
 --> tests/ui/non_string_alias.rs:6:38
  |
6 | #[derive(SerializeLabeledStringEnum, DeserializeLabeledStringEnum)]
  |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `DeserializeLabeledStringEnum` (in Nightly builds, run with -Z macro-backtrace for more info)
